chartfx-generate/src/main/java/de/gsi/generate/CodeGenerator.java

```diff
@@ -0,0 +1,349 @@
+ *
+ * //// codegen: skip all
+ * //// codegen: skip int, short, long
+ * This comment at the end of a line skips all replacing on this line and just copies the line as is.
+ *
+ * //// codegen: returncast all
+ * Adds a cast to output type after a return or assignment. Note that you might have to add parentheses.
+ *
+ * //// codegen: subst:float,double:fnInteger:fnFloating
+ * //// codegen: subst%all%a%a &lt; 0 ? -a : a
+ * Performs a simple search and replace if the type matches. The separator can be chosen arbitrarily.
+ *
+ * Multiple comment flags can be given by separating them with four slashes:
+ * //// codegen: returncast short //// subst%all%a%a &lt; 0 ? -a : a
+ *
+ * @author ennerf
+ * @author Alexander Krimm
+ */
+@Mojo(name = "generate-sources", defaultPhase = LifecyclePhase.GENERATE_SOURCES)
+public class CodeGenerator extends AbstractMojo {
+    public static final String JAVA_FILE_SUFFIX = ".java";
+    private static final String PROTO_START = "//// codegen container:";
+    private static final String LINE_COMMAND = "//// codegen:";
+    private static final String TYPE_CONVERSION_SEP = "->";
+    private static final String LINE_COMMAND_SEP = "////";
+    public static final String START_TEMPLATE = "//// codegen:";
+    public static final String END_TEMPLATE = "//// end codegen";
+    public static final String COMMAND_SKIP = "skip";
+    public static final String COMMAND_CAST_RETURN = "returncast";
+    public static final String COMMAND_SUBST = "subst";
+    public static final String PROTO_UTILCLASS_SUFFIX = "Gen";
+    public static final String PROTOTYPE_CLASS_SUFFIX = "Proto";
+    public static final String AUTOGENERATED_WARNING_HEADER = "// This file has been generated automatically by chartfx-generate. Do not modify!\n";
+
+    @Parameter(defaultValue = "${project}", required = true, readonly = true)
+    MavenProject project;
+    @Parameter(defaultValue = "${project.basedir}/src/main/codegen", readonly = true)
+    String input;
+    @Parameter(defaultValue = "${project.build.directory}/generated-sources/codegen", readonly = true)
+    String output;
+
+    @Override
+    public void execute() throws MojoExecutionException {
+        final Path inputPath = Path.of(this.input);
+        final Path outputPath = Path.of(this.output);
+
+        // Add the generated classes to the build path
+        project.addCompileSourceRoot(this.output);
+        if (getLog().isInfoEnabled()) {
+            getLog().info("Added directory for generated sources to compile sources: " + outputPath);
+        }
+
+        // Adding other types to helper function classes (...GenBase.java)
+        extendUtilityClass(inputPath, outputPath);
+
+        // Adding new classes for different types (...DoubleProto.");java)
+        addDataTypeVariants(inputPath, outputPath);
+    }
+
+    private void addDataTypeVariants(final Path inputPath, final Path outputPath) throws MojoExecutionException {
```

Comment:
Codacy found an issue: [The method 'addDataTypeVariants(Path, Path)' has an NPath complexity of 2002, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5780327)